scheduler/workflow.go

```diff
@@ -431,7 +452,12 @@ func submitPublishJob(pj job, t *task, wg *sync.WaitGroup, pu *publishNode) {
 	// Decrement the waitgroup
 	defer wg.Done()
 	// Create a new process job
-	j := newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.metricsManager, t.id)
+	var j job
+	if pu.Hostname != "" {
+		j = newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.RemoteManagers.Get(pu.Hostname), t.id)
```

Comment:
In this implementation that action is what is happening. t.metricManager is the local controlproxy that currently is talking to the local instance.

The reason for keeping the idea of a local vs remote separate is to allow us to easily move to talking to local instance in-process instead of over grpc. Right now I think that would require changing ~2 lines in snapd. I think that change would make sense since then we wouldn't have to impose the overhead of grpc on tasks that happen completely locally.